armonik/outputs.tf

```
@@ -1,3 +1,52 @@
+locals {
+  # Local variables to determine endpoint details
+  tls_enabled  = var.ingress != null ? var.ingress.tls : false
+  mtls_enabled = var.ingress != null ? var.ingress.mtls : false
+
+  # Determine endpoint and protocol
+  grpc_protocol = local.tls_enabled ? "https" : "http"
+  endpoint_url = var.ingress != null ? (
+    "${local.grpc_protocol}://${module.ingress_endpoint.ip}:${var.ingress.grpc_port}"
+  ) : "${module.control_plane_endpoint.ip}:${var.control_plane.port}"
+
+  # Base configuration map with endpoint
+  armonik_config_map = {
+    endpoint = local.endpoint_url
+  }
+
+  # Add certificates based on TLS/mTLS configuration
+  armonik_config_with_certs = local.tls_enabled ? (
+    local.mtls_enabled && var.ingress.generate_client_cert ? merge(
+      local.armonik_config_map,
+      {
+        certificate_authority = abspath("${path.root}/generated/certificates/ingress/ca.crt")
+        client_certificate    = abspath("${path.root}/generated/certificates/ingress/client.submitter.crt")
+        client_key            = abspath("${path.root}/generated/certificates/ingress/client.submitter.key")
+      }
+      ) : merge(
+      local.armonik_config_map,
+      {
+        certificate_authority = abspath("${path.root}/generated/certificates/ingress/ca.crt")
+      }
+    )
+  ) : local.armonik_config_map
```

You can heavily simplify this be leveraging what is already computed in the rest of the module

```suggestion
  armonik_cli_config = {
    endpoint              = coalesce(local.ingress_grpc_url, local.control_plane_url)
    certificate_authority = one(local_sensitive_file.ingress_ca[*].filename)
    client_certificate    = try(local_sensitive_file.ingress_client_crt["submitter"].filename, null)
    client_key            = try(local_sensitive_file.ingress_client_key["submitter"].filename, null)
  }
```

Values can be null with this design, but we can filter them out afterwards.